src/main/java/project/bookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package project.bookstore.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import project.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import project.bookstore.dto.category.CategoryDto;
+import project.bookstore.dto.category.CreateCategoryRequestDto;
+import project.bookstore.exception.unchecked.EntityNotFoundException;
+import project.bookstore.mapper.CategoryMapper;
+import project.bookstore.model.Category;
+import project.bookstore.repository.book.BookRepository;
+import project.bookstore.repository.category.CategoryRepository;
+import project.bookstore.service.CategoryService;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = categoryRepository.save(categoryMapper.toModel(requestDto));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto findCategoryById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find category by id " + id)
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find category by id " + id)
+        );
+        categoryMapper.updateBookFromDto(category, requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        if (categoryRepository.existsById(id)) {
+            categoryRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Can`t delete category by id " + id);
+        }
```

Comment:
```suggestion
        if (!categoryRepository.existsById(id)) {
          throw new EntityNotFoundException("Can`t delete category by id " + id);  
        } 
        categoryRepository.deleteById(id);
```
But in general, you can remove this check.